src/libraries/System.Private.CoreLib/src/System/Environment.GetFolderPathCore.Unix.cs

```diff
@@ -85,31 +85,23 @@ private static string GetFolderPathCoreWithoutValidation(SpecialFolder folder)
             switch (folder)
             {
                 case SpecialFolder.UserProfile:
-                case SpecialFolder.MyDocuments: // same value as Personal
                     return home;
-                case SpecialFolder.ApplicationData:
-                    return GetXdgConfig(home);
-                case SpecialFolder.LocalApplicationData:
-                    // "$XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored."
-                    // "If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used."
-                    string? data = GetEnvironmentVariable("XDG_DATA_HOME");
-                    if (string.IsNullOrEmpty(data) || data[0] != '/')
-                    {
-                        data = Path.Combine(home, ".local", "share");
-                    }
-                    return data;
-
                 case SpecialFolder.Desktop:
                 case SpecialFolder.DesktopDirectory:
                     return ReadXdgDirectory(home, "XDG_DESKTOP_DIR", "Desktop");
                 case SpecialFolder.Templates:
                     return ReadXdgDirectory(home, "XDG_TEMPLATES_DIR", "Templates");
-                case SpecialFolder.MyVideos:
-                    return ReadXdgDirectory(home, "XDG_VIDEOS_DIR", "Videos");
-
 #if TARGET_OSX
+                case SpecialFolder.ApplicationData:
+                    return Path.Combine(home, "Library", "Application Support");
+                case SpecialFolder.LocalApplicationData:
+                    return Path.Combine(home, "Library");
```

Comment:
Apps store their settings in `ApplicationData` and `LocalApplicationData`. Changing the locations will cause apps to no longer find their settings.

With no access to the previous location, implementing setting migration is also difficult.

Besides it being more correct, is there a tangible improvement?

Reply:
> With no access to the previous location, implementing *settings migration* is also difficult.

The previous location was just `/home/.config` / `/home/.local/share` so that can be used for migration. Would need to be made clear for a release.

> is there a tangible improvement?

It follows Apple's Guidelines, which from what I know are necessary to fulfill in order to publish anything on their App Store.

Reply:
> It follows Apple's Guidelines, which from what I know are necessary to fulfill in order to publish anything on their App Store.

Without this change, .NET apps aren't accepted in the Apple App Store?

Reply:
Not as far as I know, but am also not a macOS developer so never dealt with that in any way.
[Based on](https://developer.apple.com/app-store/review/guidelines/#before-you-submit) :
> Check whether your app follows guidance in other documentation

I'd assume so at least.

Reply:
Is there further information on using `~/Library` for `LOCAL_APP_DATA`?

I have used `~/Library/Application Support` for that as well (https://github.com/dirs-dev/directories-jvm#basedirectories) and haven't seen issues or user complaints.

Reply:
I based this mainly off [this StackOverflow post](https://stackoverflow.com/a/5084892) which in turn used this [rather old now Mac Java FAQ](https://developer.apple.com/library/archive/qa/qa1170/_index.html) and maybe the fact that the [iOS doc mentions](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW12) creating custom subdirectories.
I do however see people use `~/Library/Application Support` for [quite](https://doc.qt.io/qt-5/qstandardpaths.html) [a few](https://github.com/sindresorhus/env-paths) [other](https://github.com/adrg/xdg) [libs](https://github.com/sago007/PlatformFolders). So probably a good call to indeed change it to that instead.

Reply:
@tmds

> backwards compatibility

Having a good migration story is important, but we can't keep pretending that accreting technical debt instead of fixing things is fine. That's one of the things that brought computing into disrepute in the wider public.

Perhaps the behavior could be predicated on the .NET version targeted, such that only code compiled for the next version gets the correct folders?

>  It will break things, and the benefit is not clear to me (besides correctness).

Writing things to `$HOME` (e. g. `~/.dotnet`) also breaks things. You cannot expect `$HOME` to be writable. (Example: I make `$HOME` read-only on every machine under my control.)

Reply:
For any breaking change you need to weigh the pros and cons. Calling it _technical debt_ doesn't change that.

Reply:
Yes, and?

Reply:
@tannergooding Agreed. If special folders are finally updated, let's make sure they are actually correct. Those who incorrectly relied on buggy behavior should make a suggestion on how to address their problems, not the people fixing things.